WireDomain/Sources/WireDomain/Storage/UpdateEventsUserDefaults/UpdateEventsUserDefaults.swift

```diff
@@ -0,0 +1,43 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+public enum UpdateEventsUserDefaults: UserDefaultsProtocol {
+    public nonisolated(unsafe) static var userID: UUID!
+
+    enum Key: String {
+        case lastEventID
+    }
+
+    @UserDefault(key: Key.lastEventID, userID: userID)
```

Comment:
I would expect to see this declared inside the `UpdateEventRepository`. Is there a particular reason to wrap it in the enum?

Reply:
The point of the enum is to provide a **scoped** UserDefaults API depending on the domain concern. For instance, we could have a `enum ConversationUserDefaults` API or a `enum FederationUserDefaults` (with related defaults operations) API exposed publicly by WireDomain so clients could consume it in a simple and intuitive way. This helps make a clear separation of concerns between different defaults storage responsibilities.